sklearn/ensemble/_forest.py

```diff
@@ -315,6 +315,12 @@ def fit(self, X, y, sample_weight=None):
         # Remap output
         self.n_features_ = X.shape[1]
 
+        # add input validation for poisson criterion
+        if (self.criterion == "poisson" and np.any(y < 0)):
+            raise ValueError("Some value(s) of y are negative which is"
+                             " not allowed for Poisson regression."
+                             )
```

Comment:
```suggestion
        if self.criterion == "poisson":
                if np.any(y < 0):
                    raise ValueError("Some value(s) of y are negative which is"
                                     " not allowed for Poisson regression.")
                if np.sum(y) <= 0:
                    raise ValueError("Sum of y is not positive which is "
                                     "necessary for Poisson regression.")
```

Reply:
I think the later error would make more sense if we emphasized that the sum of y needs to be strictly positive. Alternatively we can could say that `y` cannot be uniformly zero.